util-core/src/main/java/com/twitter/util/ExceptionalJavaConsumer.java

```diff
@@ -0,0 +1,10 @@
+package com.twitter.util;
+
+/**
+ * Used for Java 8 interop.
+ * This is a SAM version of c.t.u.ExceptionalFunction (returning Unit/void).
+ */
+// @FunctionalInterface
+public interface ExceptionalJavaConsumer<T> {
```

Comment:
`@FunctionalInterface` is just a marker annotation, intended to inform the reader that the intention of the interface is to act as a SAM. With the Java 8 compiler, it also does some compile-time checks to ensure you can use the interface as a SAM (it should have only one non-default method).
For the lambda functionality itself, it has no consequences, since lambda annotations also work with non-annotated interfaces. It's only there to communicate intent.
I've commented it out since the annotation itself only exists in the Java 8 library (so that twitter-util is compatible with Java 7)